drivers/modem/gsm_ppp.c

```diff
@@ -1172,13 +1229,32 @@ void gsm_ppp_stop(const struct device *dev)
 	net_if_l2(iface)->enable(iface, false);
 
 	if (IS_ENABLED(CONFIG_GSM_MUX)) {
+		gsm_soft_reset(gsm);
 		/* Lower mux_enabled flag to trigger re-sending AT+CMUX etc */
 		gsm->mux_enabled = false;
 
 		if (gsm->ppp_dev) {
 			uart_mux_disable(gsm->ppp_dev);
 		}
 	}
+	else {
+		if (gsm->setup_done) {
```

Comment:
If the setup is done if the muxing is not enabled, modem would be in data mode and AT commands won't be accessible. First we need to exit PPP mode and then execute soft reset using AT channel.